test/Frontend/Desugar.hs

```diff
@@ -1,59 +1,101 @@
 module Frontend.Desugar where
 
-import Data.Attoparsec.ByteString
-import qualified Data.Attoparsec.ByteString.Char8 as Char8
+import Data.Attoparsec.ByteString (parse)
+import Frontend.Parser (shouldParseAs)
 import qualified Juvix.Frontend.Parser as Parser
-import Juvix.FrontendDesugar
-import Juvix.Library hiding (show)
+import Juvix.Frontend.Types (TopLevel)
+import Juvix.Frontend.Types.Base
+  ( Cond' (C'),
+    CondLogic'
+      ( CondExpression',
+        annCondExpression,
+        condLogicBody,
+        condLogicPred
+      ),
+    Constant' (Number'),
+    Expression' (Constant', Infix', Name'),
+    Function' (Func'),
+    FunctionLike'
+      ( Like',
+        annLike,
+        functionLikeArgs,
+        functionLikeBody,
+        functionLikedName
+      ),
+    GuardBody' (Guard'),
+    Infix' (Inf', annInf, infixLeft, infixOp, infixRight),
+    Numb' (Integer''),
+    TopLevel' (Function'),
+  )
+import Juvix.Library
+  ( Alternative (many),
+    ByteString,
+    NonEmpty ((:|)),
+    Symbol (Sym),
+  )
 import qualified Test.Tasty as T
-import qualified Test.Tasty.HUnit as T
-import Prelude (String, show)
 
 allDesugar :: T.TestTree
 allDesugar =
   T.testGroup
     "desugar Tests"
-    [ guardTest
-    ]
+    [guardTest]
 
-desugarTasty ::
-  (Show a1, Show a2, Eq a1) => T.TestName -> Either a1 a2 -> String -> T.TestTree
-desugarTasty name x y =
-  T.testGroup
-    "desugar Test"
-    [ T.testCase (name <> " parser failed") (isRight x T.@=? True),
-      T.testCase
-        ("desugar: " <> name <> " " <> " should desugar to " <> y)
-        (fmap show x T.@=? Right y)
-    ]
-
-shouldDesugar :: T.TestName -> ByteString -> String -> T.TestTree
-shouldDesugar name parseString =
-  desugarTasty name (fmap desugar (parseOnly (many Parser.topLevelSN) parseString))
+shouldDesugar ::
+  T.TestName -> ByteString -> [TopLevel] -> T.TestTree
+shouldDesugar name =
+  shouldParseAs name (parse (many Parser.topLevelSN))
```

Comment:
I'm not sure the desugar test should do this...  I don't see where we desugar

Reply:
We should be calling `desugar` somewhere, right?

Reply:
Oops corrected.